app/controllers/users_controller.rb

```diff
@@ -0,0 +1,31 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    return render "users/show" if @user
+
+    flash[:danger] = t(".not_found")
+    redirect_to root_path
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      flash[:success] = t(".new.success")
+      redirect_to @user
+    else
+      render "users/new"
```

Comment:
Thêm flash thông báo lỗi em hỉ
```suggestion
      render :new
```

Reply:
Cái thông báo lỗi mình lấy từ `@user.errors` ra rồi hiển thị ở đây nên chắc k cần `flash` ở `controller` nữa c nhỉ
https://github.com/trinhdvt/sample_app/blob/97edcdc573649be0a22299ba8124250e1e8d9d58/app/views/shared/_error_messages.html.erb#L4